lld/test/ELF/linkerscript/symbolreferenced.s

```diff
@@ -40,13 +51,23 @@ patatino:
   movl newsym, %eax
 
 #--- chain.t
-PROVIDE(f2 = 0x1000);
+PROVIDE(f7 = 0x1000);
+PROVIDE(f5 = f6);
```

Comment:
can be more complex by adding some `+1`, for example.

Reply:
Add a cycle example to either this test or the new test?

Reply:
I have made the test more complex and added a new test with cycle example.